### PR TITLE
Create Other Clinic schema 🗺️  $ Schema Design📸 

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR We present Creating another Clinic schema 🗺️  $ Schema Design📸 
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
- use GitFlow 🌊.
- Create new `schema_based_on_diagram.sql` file where you implement the database from the diagram.. ⛱️
- Join tables from many-to-many relationships 🧑‍🤝‍🧑 might not appear in the diagram 🏠 , but you still need them.
- Add the FK indexes🗝️ .
>Schema Diagram
new_clinic_schema_design
## Achievements
- use GitFlow 🌊.✅
- Create new `schema_based_on_diagram.sql` file where you implement the database from the diagram.. ⛱️ ✅ 
- Join tables from many-to-many relationships 🧑‍🤝‍🧑 might not appear in the diagram 🏠 , but you still need them. ✅
- Add the FK indexes🗝️ .✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
+CREATE TABLE "treatments"(
+    "id" INTEGER NOT NULL,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "treatments" ADD PRIMARY KEY("id");
+CREATE TABLE "medical_histories"(
+    "id" INTEGER NOT NULL,
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] Kindly consider adding `PRIMARY KEY` in this line as this id should be considered the primary key of this table. I can see that you have Altered this table and did the same thing on line 16 , but I think the concise and better way is to add the primary key inside the table as I mentioned how to do it.

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] This is a foreign key. kindly consider to change this line to: 
`patient_id int references patients(id),` to add the foreign key constraint.

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    "status" VARCHAR(255) NOT NULL
+);
+CREATE INDEX "medical_histories_patient_id_index" ON
+    "medical_histories"("patient_id");
+ALTER TABLE
+    "medical_histories" ADD PRIMARY KEY("id");
+CREATE INDEX "medical_histories_patient_id_index" ON
+    "medical_histories"("patient_id");
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] This line is repeating so kindly consider removing it to make your codebase clean and tidy.

File: schema_based_on_diagram.sql [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    "date_of_birth" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "Patients" ADD PRIMARY KEY("id");
+CREATE TABLE "invoice_items"(
+    "id" INTEGER NOT NULL,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] This is a foreign key. kindly consider to change this line to: 
`invoice_id int references invoices(id),` to add the foreign key constraint.

File: schema_based_on_diagram.sql [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    "treatment_id" INTEGER NOT NULL
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] This is a foreign key. kindly consider to change this line to: 
`treatment_id int references treatments(id),` to add the foreign key constraint.

File: schema_based_on_diagram.sql [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    "invoice_items"("treatment_id");
+ALTER TABLE
+    "invoice_items" ADD PRIMARY KEY("id");
+CREATE INDEX "invoice_items_invoice_id_index" ON
+    "invoice_items"("invoice_id");
+CREATE INDEX "invoice_items_treatment_id_index" ON
+    "invoice_items"("treatment_id");
+CREATE TABLE "medical_treatments"(
+    "medical_histories_id" INTEGER NOT NULL,
+    "treatments_id" INTEGER NOT NULL
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] These two are foreign keys. so you need to reference them here as this is a join table. kindly consider changing these lines to the following:
```
medical_history_id int references medical_histories(id),
treatment_id int references treatments(id),
```

File: schema_based_on_diagram.sql [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+);
+CREATE INDEX "medical_treatments_medical_histories_id_index" ON
+    "medical_treatments"("medical_histories_id");
+CREATE INDEX "medical_treatments_treatments_id_index" ON
+    "medical_treatments"("treatments_id");
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] The foreign key constraints should be in singular name form so consider changing these two ids in brackets to `medical_history_id` and `treatment_id`

File: schema_based_on_diagram.sql [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+CREATE INDEX "medical_treatments_medical_histories_id_index" ON
+    "medical_treatments"("medical_histories_id");
+CREATE INDEX "medical_treatments_treatments_id_index" ON
+    "medical_treatments"("treatments_id");
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] These lines are repeating so consider removing them to make your codebase clean.

File: schema_based_on_diagram.sql [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+CREATE TABLE "Invoices"(
+    "id" INTEGER NOT NULL,
+    "total_amount" DECIMAL(8, 2) NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] This is a foreign key. kindly consider to change this line to: 
`medical_history_id int references medical_histories(id),` to add the foreign key constraint.

File: schema_based_on_diagram.sql [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    "Invoices"("medical_history_id");
+ALTER TABLE
+    "medical_treatments" ADD CONSTRAINT "medical_treatments_treatments_id_foreign" FOREIGN KEY("treatments_id") REFERENCES "treatments"("id");
+ALTER TABLE
+    "medical_histories" ADD CONSTRAINT "medical_histories_patient_id_foreign" FOREIGN KEY("patient_id") REFERENCES "Patients"("id");
+ALTER TABLE
+    "medical_treatments" ADD CONSTRAINT "medical_treatments_medical_histories_id_foreign" FOREIGN KEY("medical_histories_id") REFERENCES "medical_histories"("id");
+ALTER TABLE
+    "invoice_items" ADD CONSTRAINT "invoice_items_treatment_id_foreign" FOREIGN KEY("treatment_id") REFERENCES "treatments"("id");
+ALTER TABLE
```
Comment (Zuhaib042) [inline schema_based_on_diagram.sql]: - [ ] Here you are adding the foreign key constraint where needed but the problem is that we need to add the indexes on all the foreign keys, the index adding should occur after adding the foreign keys, not before them.
If you apply the changes I suggested in the same order, then you would be adding those indexes after adding the foreign keys respectively.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @gisachris  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Well tried to implement all the requirements


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @gisachris and @toyman640 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gisachris** [comment]
Thanks for the great review @Owusu-Desmond 👏

**Pull request merged**